source/code/plugin/filter_cadvisor_health_container.rb

```diff
@@ -0,0 +1,263 @@
+#!/usr/local/bin/ruby
+# frozen_string_literal: true
+
+module Fluent
+    require 'logger'
+    require 'json'
+    require_relative 'oms_common'
+    require_relative 'HealthMonitorUtils'
+    require_relative 'HealthMonitorState'
+    require_relative "ApplicationInsightsUtility"
+
+
+    class CAdvisor2ContainerHealthFilter < Filter
+        Fluent::Plugin.register_filter('filter_cadvisor_health_container', self)
+
+        config_param :log_path, :string, :default => '/var/opt/microsoft/docker-cimprov/log/health_monitors.log'
+        config_param :metrics_to_collect, :string, :default => 'cpuUsageNanoCores,memoryRssBytes'
+        config_param :container_resource_refresh_interval_minutes, :integer, :default => 5
+
+        @@object_name_k8s_node = 'K8SNode'
+        @@object_name_k8s_container = 'K8SContainer'
+
+        @@counter_name_cpu = 'cpuusagenanocores'
+        @@counter_name_memory_rss = 'memoryrssbytes'
+
+        @@health_monitor_config = {}
+
+        @@hostName = (OMS::Common.get_hostname)
+        @@clusterName = KubernetesApiClient.getClusterName
+        @@clusterId = KubernetesApiClient.getClusterId
+        @@clusterRegion = KubernetesApiClient.getClusterRegion
+        @@cluster_health_model_enabled = HealthMonitorUtils.is_cluster_health_model_enabled
+
+        def initialize
+            super
+            @cpu_capacity = 0.0
+            @memory_capacity = 0.0
+            @last_resource_refresh = DateTime.now.to_time.to_i
+            @metrics_to_collect_hash = {}
+        end
+
+        def configure(conf)
+            super
+            @log = HealthMonitorUtils.getLogHandle
+            @log.debug {'Starting filter_cadvisor2health plugin'}
+        end
+
+        def start
+            super
+            @metrics_to_collect_hash = HealthMonitorUtils.build_metrics_hash(@metrics_to_collect)
+            @log.debug "Calling ensure_cpu_memory_capacity_set cpu_capacity #{@cpu_capacity} memory_capacity #{@memory_capacity}"
+            node_capacity = HealthMonitorUtils.ensure_cpu_memory_capacity_set(@@hm_log, @cpu_capacity, @memory_capacity, @@hostName)
+            @cpu_capacity = node_capacity[0]
+            @memory_capacity = node_capacity[1]
+            @log.info "CPU Capacity #{@cpu_capacity} Memory Capacity #{@memory_capacity}"
+            #HealthMonitorUtils.refresh_kubernetes_api_data(@log, @@hostName)
+            @@health_monitor_config = HealthMonitorUtils.getHealthMonitorConfig
+            ApplicationInsightsUtility.sendCustomEvent("filter_cadvisor_health Plugin Start", {})
```

Comment:
not sure if any of the start() events are useful at all. Have a rescue and log an exception when u handle. T

Reply:
ignore this plugin for this CR. It is only present in source control